scripts/west_commands/runners/ezflashcli.py

```diff
@@ -45,7 +45,13 @@ def program_bin(self):
             self.check_call([self.tool] + self.sn_arg + ["erase_flash"])
 
         self.logger.info(f"Flashing {self.bin_}...")
-        self.check_call([self.tool] + self.sn_arg + ["image_flash", self.bin_])
+        # 'image_flash' writes image and a valid product header which is required by internal
+        # bootloader to run application image. 'write_flash' simply writes image to given offset.
+        if self.build_conf.getboolean('CONFIG_SOC_DA1469X_WRITE_PRODUCT_HEADER'):
```

Comment:
Sorry if I missed some previous discussion in earlier iterations of the patch, but I'm not sure what the value of this symbol is. Since this option is promptless, would it make sense to inline the logic in soc/arm/renesas_smartbond/da1469x/Kconfig.soc into this file?

I.e. write the product header unless `USE_DT_CODE_PARTITION && !MCUBOOT`, which can also be detected from `self.build_conf`, without using the intermediate option. It seems like a lot of SoCs are supported by this tool, so my concern is that this will become a messy list of soc-specific if/else here.

Reply:
Yes, that makes sense. I removed option and updated script with proper logic and comments.